comms/src/outbound_message_service/outbound_message_pool/message_retry_service.rs

```diff
@@ -0,0 +1,203 @@
+//  Copyright 2019 The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use super::error::OutboundMessagePoolError;
+use crate::{
+    connection::{Connection, Direction, EstablishedConnection, InprocAddress, ZmqContext},
+    message::FrameSet,
+    outbound_message_service::{outbound_message_pool::OutboundMessagePoolConfig, OutboundMessage},
+};
+use log::*;
+use std::{
+    collections::BinaryHeap,
+    sync::mpsc::{Receiver, RecvTimeoutError},
+    thread::{self, JoinHandle},
+    time::Duration,
+};
+use tari_utilities::message_format::MessageFormat;
+
+const LOG_TARGET: &'static str = "comms::outbound_message_service::outbound_message_pool::message_retry_pool";
+const THREAD_STACK_SIZE: usize = 512_000;
```

Comment:
this seems fairly large?

Reply:
Also should probably move both const to a types files

Reply:
It probably is large, this was done before @neonknight64 's PR, at the time I just put a value there - since all the thread consts are in files, I'm keeping it like it is - can move it out in another PR if that's the right thing to do